lib/widgets/message_list.dart

```diff
@@ -511,60 +533,93 @@ class _UnreadMarker extends StatelessWidget {
   }
 }
 
-class StreamTopicRecipientHeader extends StatelessWidget {
-  const StreamTopicRecipientHeader({super.key, required this.message});
+class StreamMessageRecipientHeader extends StatelessWidget {
+  const StreamMessageRecipientHeader({super.key, required this.message, required this.showStream});
 
   final StreamMessage message;
+  final bool showStream;
 
   @override
   Widget build(BuildContext context) {
+    // For design specs, see:
+    //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=538%3A20849&mode=dev
+    //   https://github.com/zulip/zulip-mobile/issues/5511
     final store = PerAccountStoreWidget.of(context);
 
     final stream = store.streams[message.streamId];
-    final streamName = stream?.name ?? message.displayRecipient; // TODO(log) if missing
     final topic = message.subject;
-
-    final subscription = store.subscriptions[message.streamId];
-    final streamColor = Color(subscription?.color ?? 0x00c2c2c2);
+    final swatch = store.subscriptions[message.streamId]?.colorSwatch()
+      ?? _kFallbackStreamColorSwatch;
     final contrastingColor =
-      ThemeData.estimateBrightnessForColor(streamColor) == Brightness.dark
+      (ThemeData.estimateBrightnessForColor(swatch.barBackground) == Brightness.dark)
         ? Colors.white
         : Colors.black;
+    final textStyle = TextStyle(
+      color: contrastingColor,
+      fontFamily: 'Source Sans 3',
+      fontSize: 16,
+      letterSpacing: 0.02 * 16,
+      height: (18 / 16),
+    ).merge(weightVariableTextStyle(context, wght: 600, wghtIfPlatformRequestsBold: 900));
+
+    final streamWidget = (showStream)
+      ? GestureDetector(
+          onTap: () => Navigator.push(context,
+            MessageListPage.buildRoute(context: context,
+              narrow: StreamNarrow(message.streamId))),
+          child: Row(children: [
```

Comment:
This needs `CrossAxisAlignment.baseline`, just like the other Row below. Otherwise the vertical alignment gets messed up — it's most conspicuous if you crank up your text size in system settings:
![image](https://github.com/zulip/zulip-flutter/assets/28173/3681586a-3a95-473a-991c-1876bb238688)